app/src/main/java/com/mapbox/maps/testapp/examples/LargeGeojsonPerformanceActivity.kt

```diff
@@ -0,0 +1,139 @@
+package com.mapbox.maps.testapp.examples
+
+import android.os.Bundle
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.geojson.Feature
+import com.mapbox.geojson.FeatureCollection
+import com.mapbox.geojson.LineString
+import com.mapbox.geojson.Point
+import com.mapbox.maps.CameraOptions
+import com.mapbox.maps.Style
+import com.mapbox.maps.extension.style.layers.addLayer
+import com.mapbox.maps.extension.style.layers.generated.lineLayer
+import com.mapbox.maps.extension.style.sources.addSource
+import com.mapbox.maps.extension.style.sources.generated.geoJsonSource
+import com.mapbox.maps.plugin.animation.MapAnimationOptions
+import com.mapbox.maps.plugin.animation.flyTo
+import com.mapbox.maps.testapp.R
+import com.mapbox.maps.testapp.examples.annotation.AnnotationUtils
+import kotlinx.android.synthetic.main.activity_large_geojson.*
+
+/**
+ * Example of passing large geojson to verify it does not block UI thread.
+ */
+class LargeGeojsonPerformanceActivity : AppCompatActivity() {
+
+  private lateinit var routePoints: FeatureCollection
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_large_geojson)
+
+    routePoints = FeatureCollection.fromFeature(
+      Feature.fromGeometry(
+        LineString.fromPolyline(
+          DirectionsResponse.fromJson(
+            AnnotationUtils.loadStringFromAssets(
+              this@LargeGeojsonPerformanceActivity, "long_route.json"
+            )
+          ).routes()[0].geometry()!!,
+          6
+        )
+      )
+    )
+
+    mapView.getMapboxMap()
+      .apply {
+        setCamera(
+          CameraOptions.Builder()
+            .center(Point.fromLngLat(LONGITUDE, LATITUDE))
+            .zoom(START_ZOOM)
+            .build()
+        )
+        loadStyleUri(Style.MAPBOX_STREETS) { style ->
+          buttonLoad.setOnClickListener {
+            flyTo(
+              CameraOptions.Builder()
+                .zoom(END_ZOOM)
+                .build(),
+              MapAnimationOptions.mapAnimationOptions {
+                duration(10_000L)
+              })
+            it.postDelayed({
+              loadGeoJson(style)
+            }, 1000)
+          }
+        }
+      }
+  }
+
+  private fun loadGeoJson(style: Style) {
+    val start = System.currentTimeMillis()
+
+    // SYNC METHOD
+
+//    style.addSource(geoJsonSource(SOURCE) {
+//      featureCollection(routePoints)
+//    })
+
+    // ASYNC METHOD
+
+    geoJsonSource(
+      id = SOURCE,
+      block = {
+        featureCollection(routePoints)
+      },
+      result = {
+        style.addSource(it)
+        style.addLayer(lineLayer(LAYER, SOURCE) {
+          lineColor("blue")
+        })
+      }
+    )
+
+    // add some useless extra work to parse same geojson
```

Comment:
Test app acts as reference documentation for end-users. I would prefer making the geojson file itself bigger vs adding constructs like this. We can't use this code as documentation on mapbox.com. We could however move this to a test? or even the app-extension-style.

Reply:
Of course this will be revisited, I was just adding it as a quick workaround to add much work on geojson parsing.

Reply:
> We could however move this to a test? or even the app-extension-style.

This sounds like a good instrumentation test in `the app-extension-style`.